test/apis/authentication-api.ts

```diff
@@ -5,13 +5,97 @@ import "isomorphic-fetch";
 import {getEnvVariables, setLocalStorage, TestUtils} from "../test-utils";
 import {AnonymousMedTechApiBuilder} from "../../src/apis/AnonymousMedTechApi";
 import {webcrypto} from "crypto";
+import {MedTechApiBuilder} from "../../src/apis/medTechApi";
 
 setLocalStorage(fetch);
 
 const {iCureUrl: iCureUrl, msgGtwUrl: msgGtwUrl, authProcessHcpId: authProcessHcpId
 } = getEnvVariables()
 
 describe("Authentication API", () => {
+
+  it("AnonymousMedTechApi shouldn't be instantiated if both authServerUrl and specId aren't passed", async () => {
+    const authProcessId = process.env.ICURE_TS_TEST_HCP_AUTH_PROCESS_ID ??
+      "6a355458dbfa392cb5624403190c6a19"; // pragma: allowlist secret
+
+    try {
+      await new AnonymousMedTechApiBuilder()
+        .withICureUrlPath(iCureUrl)
+        .withAuthServerUrl(msgGtwUrl)
+        .withCrypto(webcrypto as any)
+        .withAuthProcessId(authProcessId)
+        .build();
+      expect(true, "promise should fail").eq(false)
+    } catch (e) {
+      expect((e as Error).message).to.eq("authSpecId is required")
+    }
+
+    try {
+      const anonymousMedTechApi = await new AnonymousMedTechApiBuilder()
+        .withICureUrlPath(iCureUrl)
+        .withAuthServerUrl(msgGtwUrl)
+        .withAuthSpecId(msgGtwSpecId)
+        .withCrypto(webcrypto as any)
+        .withAuthProcessId(authProcessId)
+        .build();
+
+      expect(anonymousMedTechApi, "anonymousMedTechApi shouldn't be null").not.null
+    } catch (e) {
+      expect(true, "promise should not fail").eq(false)
+    }
+
+    const anonymousMedTechApi = await new AnonymousMedTechApiBuilder()
+      .withICureUrlPath(iCureUrl)
+      .withAuthServerUrl(msgGtwUrl)
+      .withAuthSpecId(msgGtwSpecId)
+      .withCrypto(webcrypto as any)
+      .withAuthProcessId(authProcessId)
+      .build();
+  })
+
+  it("MedTechApi shouldn't be instantiated if both authServerUrl and specId aren't passed", async () => {
```

Comment:
In fact, it can be instantiate but the user should not use the authenticationApi if authServerUrl, authProcessId & authSpecId are missing. 
Otherwise, it means even if the user is authenticated, the developer always needs to initialize the authentication info which would be useless